packages/react-widgets/src/widgets/utils/sortLayersById.js

```diff
@@ -0,0 +1,16 @@
+import { assert } from '@carto/react-core';
+
+/**
+ * Sorts a list of layers based on a list of id property values
+ * @param {any[]} layers list of layers to sort
+ * @param {string[]} layerOrder values to sort by
+ * @returns sorted array
+ */
+export default function sortLayersById(layers = [], layerOrder = []) {
+  const uniqueValues = [...new Set(layerOrder)];
+  assert(
+    layers.every((layer) => layer.id),
+    'Layers must have an id property'
+  );
+  return layers.sort((a, b) => uniqueValues.indexOf(b.id) - uniqueValues.indexOf(a.id));
```

Comment:
```suggestion
  return [...layers].sort((layerA, layerB) => {
      assert(layerA.id && layerB.id, 'Layer must have an id')
      return uniqueValues.indexOf(layerB.id) - uniqueValues.indexOf(layerA.id));
  }
```

Reply:
I added `[...layers]` to avoid modifying layers original reference so `sortLayers` can be a pure fn.

Reply:
perfect! but conceptually  sort (like the Arrays.prototype function), returns the array but also modifies the order of the original ones to.

Reply:
That why I added the `[...layers]`.